harvester_e2e_tests/integration/test_rancher_integration.py

```diff
@@ -0,0 +1,319 @@
+            unique_name,
+            data['status']['clusterName'],
+            harvester_kubeconfig
+        )
+
+        assert 201 == code, (
+            f"Failed to create cloud credential with error: {code}, {data}"
+        )
+
+    def test_create_vlan(self, api_client, unique_name):
+        code, data = api_client.networks.create(unique_name, 1)
+
+        assert 201 == code, (
+            f"Failed to create vlan network with error: {code}, {data}"
+        )
+
+    def test_upload_image(self, api_client, unique_name, wait_timeout):
+        code, data = api_client.images.create_by_url(
+            unique_name,
+            'http://cloud-images.ubuntu.com/releases/focal/release/ubuntu-20.04-server-cloudimg-amd64-disk-kvm.img'  # noqa
+        )
+
+        assert 201 == code, (
+            f"Failed to upload focal image with error: {code}, {data}"
+        )
+
+        endtime = datetime.now() + timedelta(seconds=wait_timeout)
+        while endtime > datetime.now():
+            code, data = api_client.images.get(unique_name)
+            if 'status' in data and 'progress' in data['status'] and \
+                    data['status']['progress'] == 100:
+                break
+            sleep(5)
+        else:
+            raise AssertionError(
+                f"Image {unique_name} can't be ready with {wait_timeout} timed out\n"
+                f"Got error: {code}, {data}"
+            )
+
+    def test_create_rke2(self, api_client, rancher_api_client, unique_name, k8s_version):
+        code, data = rancher_api_client.mgmt_clusters.get(
+            self.get_harvester_cluster_name(unique_name)
+        )
+        clusterName = data['status']['clusterName']
+        resp = rancher_api_client.kube_configs.create(
+            self.get_rke2_cluster_name(unique_name),
+            clusterName
+        )
+        assert 200 == resp.status_code, (
+            f"Failed to create harvester kubeconfig for rke2 with error: \
+                {resp.status_code}, {resp.text}"
+        )
+
+        kubeconfig = resp.text
+        print(kubeconfig)
+
+        code, data = rancher_api_client.secrets.create(
+            name=unique_name,
+            namespace=FLEET_DEFAULT_NAMESPACE,
+            data={
+                "credential": base64.b64encode(kubeconfig.encode('UTF-8')).decode('UTF-8')
+            },
+            annotations={
+                "v2prov-secret-authorized-for-cluster": self.get_rke2_cluster_name(unique_name),
+                "v2prov-authorized-secret-deletes-on-cluster-removal": "true"
+            }
+        )
+
+        assert 201 == code, (
+            f"Failed to create secret with error: {code}, {data}"
+        )
+
+        code, data = rancher_api_client.harvester_configs.create(
+            name=unique_name,
+            cpu_count="2",
+            disk_size="40",
+            memory_size="4",
+            image_name=f"{DEFAULT_NAMESPACE}/{unique_name}",
```

Comment:
namespace should be determined by the image, not here.

here is the example: https://github.com/harvester/tests/blob/043f06eaa38e6dd114ceb10a1bbfedbd02ec2dc8/harvester_e2e_tests/integration/test_vm_functions.py#L34-L56